File: Strategy/StrategyPython/warmachine.py
from weapon import Weapon

class WarMachine :
    _weopen = None
    
    def move(self):
        print("WarMachine is moving")
    
    def set_weapon(self,weapon:Weapon):
        self._weopen = weapon
    
    def fire(self):
        self._weopen.fire()

File: Strategy/strategy-dart/war_machine.dart
import 'weapon.dart';

class WarMachine {
  late Weapon _weapon;

  set setWeapon(Weapon value) {
    _weapon = value;
  }

  void move() {
    print('Machine is Moving');
  }

  void fire() {
    _weapon.fire();
  }
}
